### PR TITLE
Always allow updating a call order to higher collateral ratio

### PR DESCRIPTION
**Pull request by abitmore**
PR for #583

### CODE REVIEW

File: libraries/chain/market_evaluator.cpp [hunk 1]
```diff
@@ -258,17 +267,39 @@ void_result call_order_update_evaluator::do_apply(const call_order_update_operat
...
-         // were no matching orders.  In the latter case, we throw.
-         GRAPHENE_ASSERT(
-            ~call_obj->call_price < _bitasset_data->current_feed.settlement_price,
-            call_order_update_unfilled_margin_call,
-            "Updating call order would trigger a margin call that cannot be fully filled",
-            ("a", ~call_obj->call_price )("b", _bitasset_data->current_feed.settlement_price)
-            );
+         if( d.head_block_time() <= HARDFORK_CORE_583_TIME )
+         {
+            //edump( (~call_obj->call_price) ("<")( _bitasset_data->current_feed.settlement_price) );
```
Comment (pmconrad) [inline libraries/chain/market_evaluator.cpp]: Please remove this commented out edump() and add a TODO to remove the check after HF has passed.

File: libraries/chain/market_evaluator.cpp [hunk 2]
```diff
@@ -198,6 +198,8 @@ void_result call_order_update_evaluator::do_apply(const call_order_update_operat
    auto itr = call_idx.find( boost::make_tuple(o.funding_account, o.delta_debt.asset_id) );
    const call_order_object* call_obj = nullptr;
 
+   price old_collateralization;
```
Comment (pmconrad) [inline libraries/chain/market_evaluator.cpp]: Perhaps make this an optional\<price> and use its truth value instead of is_null() further down?

Comment (abitmore) [reply libraries/chain/market_evaluator.cpp]: That's what I coded in the first place, however, `optional<>` will produce a warning on compile that's why I don't like it. It does provide a bit more performance though.

Comment (abitmore) [reply libraries/chain/market_evaluator.cpp]: Oh, it doesn't produce a warning here. Will make the change soon.

**Pull request merged**